### PR TITLE
add version selection

### LINKED ISSUE #299: support to config the version white list for client
https://github.com/vesoft-inc/nebula-java/issues/567

### PR DESCRIPTION
**Pull request by javaGitHub2022**
close #299

### CODE REVIEW

File: nebula3/gclient/net/ConnectionPool.py [hunk 1]
```diff
@@ -206,7 +207,7 @@ def ping(self, address):
         """
         try:
             conn = Connection()
-            conn.open_SSL(address[0], address[1], 1000, self._ssl_configs)
+            conn.open_SSL(address[0], address[1], 1000, self._configs.version,self._ssl_configs)
```
Comment (wey-gu) [inline nebula3/gclient/net/ConnectionPool.py:210]: The missing space is one of the reason fails the CI/Lint :)

```diff
- conn.open_SSL(address[0], address[1], 1000, self._configs.version,self._ssl_configs)
+ conn.open_SSL(address[0], address[1], 1000, self._configs.version, self._ssl_configs)
```

**Review by wey-gu — APPROVED** [review]

File: nebula3/gclient/net/Connection.py [hunk 2]
```diff
@@ -40,30 +40,34 @@ def __init__(self):
         self._port = None
         self._timeout = 0
         self._ssl_conf = None
+        self.version = None
```
Comment (Nicole00) [inline nebula3/gclient/net/Connection.py:43]: the config name need to be rename to handshakeKey

File: tests/test_connection.py [hunk 3]
```diff
@@ -19,16 +19,16 @@
 from nebula3.common import ttypes
 from nebula3.Exception import IOErrorException
 
-AddrIp = ['127.0.0.1', '::1']
+AddrIp = ['127.0.0.1']
 port = 9669
-
+version="3.0.0"
```
Comment (Nicole00) [inline tests/test_connection.py:24]: the config need to be rename to handshakeKey

**Comment by Nicole00** [comment]
SessionPool config need to add config handshakeKey

File: nebula3/gclient/net/SessionPool.py [hunk 4]
```diff
@@ -27,7 +27,7 @@ class SessionPool(object):
     S_OK = 0
     S_BAD = 1
 
-    def __init__(self, username, password, space_name, addresses):
+    def __init__(self, username, password, space_name, addresses, handshakeKey):
```
Comment (Nicole00) [inline nebula3/gclient/net/SessionPool.py:30]: do not add parameter for _init_.
the handshakeKey already defined in SessionPoolConfig.

File: example/SessinPoolExample.py [hunk 5]
```diff
@@ -10,15 +10,17 @@
...
 from nebula3.Config import SessionPoolConfig
 from nebula3.gclient.net import Connection
 from nebula3.gclient.net.SessionPool import SessionPool
 
+
 if __name__ == "__main__":
     ip = "127.0.0.1"
     port = 9669
-
+    handshakeKey = "3.0.0"
```
Comment (Nicole00) [inline example/SessinPoolExample.py:23]: the example does not use the config handshakeKey.

File: nebula3/gclient/net/Connection.py [hunk 6]
```diff
@@ -40,30 +40,34 @@ def __init__(self):
...
+        self.open_SSL(ip, port, timeout, handshakeKey, None)
 
-    def open_SSL(self, ip, port, timeout, ssl_config=None):
+    def open_SSL(self, ip, port, timeout, handshakeKey=None, ssl_config=None):
         """open the SSL connection
 
         :param ip: the server ip
         :param port: the server port
         :param timeout: the timeout for connect and execute
+        :param handshakeKey: the server version
```
Comment (Nicole00) [inline nebula3/gclient/net/Connection.py:62]: handshakeKey: the key for client and server handshake, make sure the handshakeKey is in the white list config of server: client_white_list

File: nebula3/gclient/net/Connection.py [hunk 7]
```diff
@@ -89,7 +93,9 @@ def open_SSL(self, ip, port, timeout, ssl_config=None):
             header_transport.open()
 
             self._connection = GraphService.Client(protocol)
-            resp = self._connection.verifyClientVersion(VerifyClientVersionReq())
+            verifyClientVersionReq = VerifyClientVersionReq()
+            verifyClientVersionReq.version = handshakeKey
```
Comment (Nicole00) [inline nebula3/gclient/net/Connection.py:97]: config the version for verifyClientVersionReq only when handshakeKey is not None.

File: nebula3/gclient/net/ConnectionPool.py [hunk 8]
```diff
@@ -290,18 +301,15 @@ def _remove_idle_unusable_connection(self):
...
                                     connection.get_address()
                                 )
                             )
                             conns.remove(connection)
                             continue
-                        if (
-                            self._configs.idle_time != 0
-                            and connection.idle_time() > self._configs.idle_time
-                        ):
+                        if self._configs.idle_time != 0:
```
Comment (Nicole00) [inline nebula3/gclient/net/ConnectionPool.py:310]: why remove the condition for idle_time?

File: tests/test_connection.py [hunk 9]
```diff
@@ -15,14 +23,15 @@
 
 AddrIp = ["127.0.0.1", "::1"]
 port = 9669
+handshakeKey = "3.0.0"
```
Comment (Nicole00) [inline tests/test_connection.py:26]: please add some tests for wrong handshakeKey value, such as "INVALID_HANDSHAKE_KEY"

File: nebula3/gclient/net/ConnectionPool.py (hunk 8 above)
Comment (javaGitHub2022) [reply nebula3/gclient/net/ConnectionPool.py]: Accidentally deleted

**Review by Nicole00 — APPROVED** [review]
LGTM

**Pull request merged**